packages/connection-form/src/components/advanced-options-tabs/authentication-tab/authentication-oidc.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useCallback } from 'react';
+import {
+  Accordion,
+  Checkbox,
+  Description,
+  FormFieldContainer,
+  Label,
+  TextInput,
+} from '@mongodb-js/compass-components';
+import type ConnectionStringUrl from 'mongodb-connection-string-url';
+import type { ConnectionOptions } from 'mongodb-data-service';
+import type { AuthFlowType } from '@mongodb-js/oidc-plugin';
```

Comment:
Do you want to infer this from `OIDCOptions` below? That would remove the dependency on `@mongodb-js/oidc-plugin` here

```suggestion
type ExtractArrayEntryType<T> = T extends (infer U)[] ? U : never;
type AuthFlowType = ExtractArrayEntryType<OIDCOptions['allowedFlows']>;
```

Reply:
This is some cool typing! Thanks, nice suggestion.